src/spell/modules/spell/spell/src/auth/mod.rs

```diff
@@ -0,0 +1,185 @@
+mod roles;
+mod keys;
+
+pub use self::roles::{is_by_spell, is_by_creator};
+
+use marine_rs_sdk::CallParameters;
+use crate::auth::keys::parse_permission;
+use crate::auth::roles::{Auth, authenticate};
+
+pub fn guard_kv_write(key: &str)  -> eyre::Result<()> {
+    let cp = marine_rs_sdk::get_call_parameters();
+    match authenticate(&cp) {
+        Auth::Spell => Ok(()),
```

Comment:
I don't understand how this works 🤔

Reply:
What exactly you don't understand?

`guard` is guarding the execution and doesn't allow to move forward when it's failed. I used to avoid copy-pasting
```
if !is_permitted() {
   return UnitValue::error("generic error message")
}
```

Reply:
I don't understand, why `Auth::Spell` means "always allowed to write".

Reply:
Well, it's not any spell; it's _the_ spell itself, rust doesn't allow to use Self as a variant. The spell itself can access everything.

Reply:
It was a small optimization to avoid parsing the key when it isn't necessary. This optimization isn't actually needed, I think, but it simplified the concept.

Reply:
Ah, okay, understood. `if role(call_parameters) == Role::Spell => can write to any key`